### DISCUSSION TITLE
Where is the k3s server config file?

### DISCUSSION THREAD

**Discussion by sdwilsh** [kairos-io/kairos#1958] [Q&A] [answered]
[k3s docs](https://docs.k3s.io/installation/configuration#configuration-file) say it should be at `/etc/rancher/k3s/config.yaml`, but that doesn't exist for Kairos.

**Comment by jimmykarily** [comment]
I will be there if k3s starts successfully. Make sure you are installing a "standard" image and not a "core" one. Also check if k3s service has started successfully. Check `systemctl status k3s` and `journalctl -u k3s` for errors.

**Reply by sdwilsh** [reply]
This particular node has been running for months, but I'll show my work :)

```
$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
KAIROS_NAME="kairos-kairos-ubuntu-22-lts"
KAIROS_VERSION="v2.4.1-k3sv1.27.3+k3s1"
KAIROS_ID="kairos"
KAIROS_ID_LIKE="kairos-kairos-ubuntu-22-lts"
KAIROS_VERSION_ID="v2.4.1-k3sv1.27.3+k3s1"
KAIROS_PRETTY_NAME="kairos-kairos-ubuntu-22-lts v2.4.1-k3sv1.27.3+k3s1"
KAIROS_BUG_REPORT_URL="https://github.com/kairos-io/kairos/issues"
KAIROS_HOME_URL="https://github.com/kairos-io/kairos"
KAIROS_IMAGE_REPO="quay.io/kairos/kairos-ubuntu-22-lts"
KAIROS_IMAGE_LABEL=""
KAIROS_GITHUB_REPO="kairos-io/provider-kairos"
KAIROS_VARIANT="kairos"
KAIROS_FLAVOR="ubuntu-22-lts"
```

The output of `sudo systemctl status k3s --no-pager -` is lengthy, so here's a gist: https://gist.github.com/sdwilsh/8177a3efc9931e9bd23845d18149922c

So k3s is running

```
$ sudo ls /etc/rancher/k3s/
k3s.yaml
```

The k3s docs also note that since `v1.21.0+k3s1`, `/etc/rancher/k3s/config.yaml.d/*.yaml` is loaded from in alphabetical order, but as we see from the `ls` output above, that folder doesn't exist.


For some additional context, I modified `/oem/90_customer.yaml` to change `k3s.args` to add `--disable-kube-proxy` in order to [enable eBPF with Calico](https://docs.tigera.io/calico/latest/operations/ebpf/enabling-ebpf).  After rebooting the node, it didn't seem to take and I'm trying to determine if kairos-provider actually updated the config.  However, I can't seem to find the config to do that, hence the original question.

**Reply by jimmykarily** [reply]
Sorry I thought you meant the kubeconfig file (k3s.yaml) which is there. I should have read more carefully. the `config.yaml` file seems optional: `In addition to configuring K3s with environment variables and CLI arguments, K3s can also use a config file.` 

As far as I can tell, provider-kairos passes the k3s args as cli args directly in the service file: https://github.com/kairos-io/provider-kairos/blob/ee4547deb9e207cdb8a336fc1e8f3dac3991be91/internal/provider/bootstrap.go#L224

Passing it through `k3s.args` sounds correct and it should put it in the systemd service file. Can you send that specific part from your Kairos config just in case the syntax is wrong or something. If we can reproduce we should move to an issue.

**Reply by sdwilsh** [reply, marked as answer]
I'll file an issue with the details and reference this discussion.  You effectively answered the original question in that it doesn't exist, but how it is invoked can be found at `/etc/systemd/system/k3s.service.d/override.conf`.
